adapters/postgres.go

```diff
@@ -479,48 +485,77 @@ func (p *Postgres) BulkInsert(table *Table, objects []map[string]interface{}) er
 }
 
 func (p *Postgres) insertInTransaction(wrappedTx *Transaction, table *Table, objects []map[string]interface{}) error {
-	var placeholders string
+	start := time.Now()
+	var placeholdersBuilder strings.Builder
 	var header []string
-	i := 1
 	for name := range table.Columns {
 		header = append(header, name)
-
-		//$1::type, $2::type, $3, etc
-		castClause := ""
-		castType, ok := p.mappingTypeCasts[name]
-		if ok {
-			castClause = "::" + castType
-		}
-		placeholders += "$" + strconv.Itoa(i) + castClause + ","
-
-		i++
 	}
-
-	query := p.insertQuery(table.GetPKFields(), table.Name, strings.Join(header, ","), removeLastComma(placeholders))
-
-	insertStmt, err := wrappedTx.tx.PrepareContext(p.ctx, query)
-	if err != nil {
-		return fmt.Errorf("Error preparing bulk insert statement [%s] table %s statement: %v", query, table.Name, err)
+	maxValues := len(objects) * len(table.Columns)
+	if maxValues > postgresValuesLimit {
+		maxValues = postgresValuesLimit
 	}
-
+	valueArgs := make([]interface{}, 0, maxValues)
+	placeholdersCounter := 1
 	for _, row := range objects {
-		var values []interface{}
-		for _, column := range header {
+		// if number of values exceeds limit, we have to execute insert query on processed rows
+		if len(valueArgs)+len(header) > postgresValuesLimit {
+			err := p.executeInsert(wrappedTx, table, header, placeholdersBuilder, valueArgs)
+			if err != nil {
+				return err
+			}
+			logging.Infof("Inserted %d rows", len(valueArgs)/len(table.Columns))
```

Comment:
This log message is useless

Reply:
+, it was there just to check how many objects are inserted in one batch